src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.cs

```diff
@@ -272,10 +277,10 @@ string EncodeUrl (Uri url)
 				if(Logger.LogNet)
 					Logger.Log (LogLevel.Info, LOG_APP, " cancelled");
 
-				httpConnection.Disconnect ();
+				await DisconnectAsync (httpConnection);
 				cancellationToken.ThrowIfCancellationRequested ();
 			}
-			cancellationToken.Register (httpConnection.Disconnect);
+			cancellationToken.Register (async () => await DisconnectAsync (httpConnection));
```

Comment:
If you want it to shift then perhaps changing it to something analogous to this would be good:
```
cancellationToken.Register (() => {
    DisconnectAsync (httpConnection).ContinueWith (t => {
        if (t.Exception != null)
            Log (t.Exception);
    }, TaskScheduler.Default);
});
```